### PR TITLE
[Chore:#162727500]Setup react

### PR DESCRIPTION
**Pull request by XROLE**
#### What does this pull request do?
- Setup .empty react project

#### Description of task to be completed?
- [x] Initialize project with package.json
- [x] install and configure webpack
- [x] install and configure babel
- [x] install and configure eslint
- [x] install and configure prettier
- [x] create a dummy component to test

#### How should this be tested?
- On the terminal, run `npm start`

#### What are the relivant pivotal tracker stories?
-  [#162727500](https://www.pivotaltracker.com/story/show/162727500)

### CODE REVIEW

**Comment by andela-mabdussalam** [comment]
Remove the `.DS_Store` files.

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+const path = require('path')
```
Comment (andela-mabdussalam) [inline webpack.config.js]: Seperate the webpack config for dev and production environment. 

You can have a base config that is extended by both the dev and prod config in a folder in the root directory.

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+// module.exports = {
```
Comment (andela-mabdussalam) [inline .eslintrc.js]: Commented code.

File: package.json [hunk 3]
```diff
@@ -1,20 +1,42 @@
 {
   "name": "pippin-ah-frontend",
   "version": "1.0.0",
-  "description": "",
+  "description": "Learngroud React",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "start": "webpack-dev-server --mode development --open --hot",
+    "build": "webpack --mode production"
```
Comment (andela-mabdussalam) [inline package.json]: We can also have separate build commands for the dev and prod environment.

File: webpack.config.js (hunk 1 above)
Comment (habibaudu) [reply webpack.config.js]: ok

File: .eslintrc.js (hunk 2 above)
Comment (habibaudu) [reply .eslintrc.js]: ok. will take them out

File: package.json (hunk 3 above)
Comment (habibaudu) [reply package.json]: ok, doing that right away

File: src/components/landingPage.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import React, { Component } from "react";
```
Comment (andela-mabdussalam) [inline src/components/landingPage.js]: Let us stick to a convention for naming components. `App.js` is pascal cased while `landingPage.js` is camel cased.

Comment (XROLE) [reply src/components/landingPage.js]: Ok noted.

File: webpack.prod.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+ const merge = require('webpack-merge');
+ const common = require('./webpack.common.js');
+
+ module.exports = merge(common, {
+   mode: 'development',
```
Comment (andela-mabdussalam) [inline webpack.prod.js]: Why `development`?

File: src/components/App.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import React, { Component} from 'react';
```
Comment (andela-mabdussalam) [inline src/components/App.js]: Do you have prettier setup correctly?

It should solve for minor issues like proper spacing.

**Comment by andela-mabdussalam** [comment]
Incase it was missed:
- Add a folder named `webpack` in the root directory, all webpack config files should be in the folder.
- Remove .DS_Store.
- Remove unnecessary spaces.
- Same naming convention should be adhered to for component names
- Commit message conventions should also be followed

Please let us get these changes out asap, especially because it is in the way of other tickets.

File: webpack.prod.js (hunk 5 above)
Comment (habibaudu) [reply webpack.prod.js]: Sorry that's a mistake, changing it now to 'production'

File: src/components/App.js (hunk 6 above)
Comment (XROLE) [reply src/components/App.js]: No but we've set it up

File: webpack/webpack.dev.js [hunk 7]
```diff
@@ -0,0 +1,6 @@
+ const merge = require('webpack-merge');
```
Comment (andela-mabdussalam) [inline webpack/webpack.dev.js:1]: I do not think this package is necessary. A spread operator would suffice.

**Comment by andela-mabdussalam** [comment]
@XROLE @habibaudu I still see two `.DS_Store` files in the `src` and `src/components` directory.

File: webpack/webpack.dev.js (hunk 7 above)
Comment (XROLE) [reply webpack/webpack.dev.js:1]: Ok. We will implement that right away

**Comment by andela-mabdussalam** [comment]
I have added `https://www.pivotaltracker.com/story/show/163721460` to remove webpack-merge.

**Review by andela-mabdussalam — APPROVED** [review]

**Pull request merged**